tests/e2e/vmservice_utils.go

```diff
@@ -338,10 +346,10 @@ func waitNgetVmsvcVmIp(ctx context.Context, c ctlrclient.Client, namespace strin
 			}
 			return false, nil
 		}
-		if vm.Status.VmIp == "" {
+		if vm.Status.Network.PrimaryIP4 == "" {
```

Comment:
`vm.Status.Network` is a pointer that can be `nil`

Reply:
Fixed.